packages/rsocket-core/src/RSocketMachine.js

```diff
@@ -888,6 +888,10 @@ class RSocketMachineImpl<D, M> implements RSocketMachine<D, M> {
       flags |= FLAGS.COMPLETE;
       this._subscriptions.delete(streamId);
     }
+    if (payload.metadata !== undefined) {
```

Comment:
```suggestion
    if (payload.metadata != undefined) {
```

I would release restriction here to make sure `null` also rejected there

Reply:
I think would need to be `payload.metadata !== undefined &&  payload.metadata !== null` as `null` won't match `undefined` even without strict equality check. Latest commit should resolve this.